include/sys/util.h

```diff
@@ -204,6 +204,13 @@ extern "C" {
 #define CLAMP(val, low, high) (((val) <= (low)) ? (low) : MIN(val, high))
 #endif
 
+/**
+ * @def IN_RANGE
+ * @brief Returns true if @p is in the @p min - @p max range (inclusive).
```

Comment:
Probably should be just "Trie if @p (...)" as it does not return anything it just evaluates an expression.

Reply:
Question: shouldn't this been split into _UINT, _INT, _FL and _DBL variants?
I think that evaluating the thing like `IN_RANGE(bu, -10, 20)`, where `bu` is `float` may trigger warnings, so will end up as: `IN_RANGE((int)bu, -10, 20)`, or `IN_RANGE(bu, (float)-10, (float)20)` anyway.

Reply:
> Also considered the name "IS_BETWEEN" as an alternative. Please let me know if that is more fitting.

`IS_BETWEEN` sound more like non-inclusive test, for example `IS_BETWEEN(1, 1, 3)` would return false because 1 is not between 1 and 3.

Reply:
> > Also considered the name "IS_BETWEEN" as an alternative. Please let me know if that is more fitting.
> 
> `IS_BETWEEN` sound more like non-inclusive test, for example `IS_BETWEEN(1, 1, 3)` would return false because 1 is not between 1 and 3.

Indeed. "IS_BETWEEN" would not be inclusive.

Reply:
> Question: shouldn't this been split into _UINT, _INT, _FL and _DBL variants? I think that evaluating the thing like `IN_RANGE(bu, -10, 20)`, where `bu` is `float` may trigger warnings, so it will end up as: `IN_RANGE((int)bu, -10, 20)`, or `IN_RANGE(bu, (float)-10, (float)20)` anyway; variants may use explicit casts to take care of that.

I did consider that, but that is up to the caller. `IN_RANGE(bu, -10.0f, 20.0f)` for `bu` being a float would be fine. 

If we go the route with types, then it should be inline functions instead IMO.